device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/Message.java

```diff
@@ -642,4 +650,16 @@ public final void setCreationTimeUTC(Date creationTimeUTC)
         // Codes_SRS_MESSAGE_34_065: [The function shall save the provided creationTimeUTC.]
         this.creationTimeUTC = creationTimeUTC;
     }
+
+    public void setAsSecurityMessage()
+    {
+        // Set the message as json encoding
+        this.contentEncoding = SECURITY_CLIENT_JSON_ENCODING;
+        this.isSecurityClient = true;
+    }
```

Comment:
Could we just re-use the API for setMessageType, and have add a security message type to that enum? This API is a little wonky, and doesn't allow users to unset it as a security message

Reply:
Yes, we had a MessageType enum before, but it got rejected because it added too many concepts to the sdk.  These are still telemetry messages.

---
In reply to: [322454736](https://github.com/Azure/azure-iot-sdk-java/pull/576#discussion_r322454736) [](ancestors = 322454736)

Reply:
The user should have to turn it off because these are specialized messages that get sent.  The user shouldn't have to unset them once they're set.

---
In reply to: [322970421](https://github.com/Azure/azure-iot-sdk-java/pull/576#discussion_r322970421) [](ancestors = 322970421,322454736)